Source/Python/WarpXWrappers.cpp

```diff
@@ -57,23 +57,23 @@ namespace
 #endif
         for ( amrex::MFIter mfi(mf, false); mfi.isValid(); ++mfi ) {
             int i = mfi.LocalIndex();
-            data[i] = (amrex::Real*) mf[mfi].dataPtr();
+            data[i] = mf[mfi].dataPtr();
             for (int j = 0; j < AMREX_SPACEDIM; ++j) {
                 (*shapes)[shapesize*i+j] = mf[mfi].box().length(j);
             }
             if (mf.nComp() > 1) (*shapes)[shapesize*i+AMREX_SPACEDIM] = mf.nComp();
         }
         return data;
     }
-    int* getMultiFabLoVects(const amrex::MultiFab& mf, int *num_boxes, int **ngrowvect)
+    int* getMultiFabLoVects (const amrex::MultiFab& mf, int *num_boxes, int **ngrowvect)
     {
         int shapesize = AMREX_SPACEDIM;
         *ngrowvect = static_cast<int*>(malloc(sizeof(int)*shapesize));
         for (int j = 0; j < AMREX_SPACEDIM; ++j) {
             (*ngrowvect)[j] = mf.nGrow(j);
         }
         *num_boxes = mf.local_size();
-        int *loVects = (int*) malloc((*num_boxes)*AMREX_SPACEDIM * sizeof(int));
+        auto loVects = static_cast<int*>(malloc((*num_boxes)*AMREX_SPACEDIM * sizeof(int)));
```

Comment:
Where are we freeing this memory again, @dpgrote ? :)

Reply:
The data is freed at the Python level. This happens in `_get_mesh_array_lovects` at the end of the routine, the line `_libc.free(data)`. That routine makes a copy of the data as a numpy array and returns that.

If you can use a `unique_ptr` or `new`/`delete` here, then there is no reason not to. The code was written as it was since it is straightforward to pass an `int*` up to Python (using ctypes).

Reply:
Ok, looks good! Thanks for clarifying :+1:

Reply:
For future reference, maybe a comment can be added here stating that this memory will be freed at the Python level. (Though I just noticed now that this was already merged.)